include/z64player.h

```diff
@@ -7,6 +7,28 @@
 
 struct Player;
 
+#define PLAYER_GET_START_MODE(thisx) PARAMS_GET_S(thisx->params, 8, 4)
+
+typedef enum PlayerStartMode {
+    /*  0 */ PLAYER_START_MODE_0,
+    /*  1 */ PLAYER_START_MODE_1,
+    /*  2 */ PLAYER_START_MODE_2,
+    /*  3 */ PLAYER_START_MODE_3,
+    /*  4 */ PLAYER_START_MODE_4,
+    /*  5 */ PLAYER_START_MODE_5,
+    /*  6 */ PLAYER_START_MODE_6,
+    /*  7 */ PLAYER_START_MODE_7,
+    /*  8 */ PLAYER_START_MODE_UNUSED_8,  // Unused, behaves the same as PLAYER_START_MODE_14
+    /*  9 */ PLAYER_START_MODE_UNUSED_9,  // Unused, behaves the same as PLAYER_START_MODE_14
+    /* 10 */ PLAYER_START_MODE_UNUSED_10, // Unused, behaves the same as PLAYER_START_MODE_14
+    /* 11 */ PLAYER_START_MODE_UNUSED_11, // Unused, behaves the same as PLAYER_START_MODE_14
+    /* 12 */ PLAYER_START_MODE_UNUSED_12, // Unused, behaves the same as PLAYER_START_MODE_14
+    /* 13 */ PLAYER_START_MODE_13,
+    /* 14 */ PLAYER_START_MODE_14,
+    /* 15 */ PLAYER_START_MODE_15,
+    /* 16 */ PLAYER_START_MODE_MAX
+} PlayerStartMode;
+
```

Comment:
Given that these are only used in `z_player`, I think that it'd be best to have them there instead of here as, otherwise, this will be included elsewhere unnecessarily.

Reply:
This is a param, it is going to be set from the scene file (and anywhere Actor_Spawn may be called)
This is true for all actors, all params should be exposed publically.

Reply:
Note that this also informs my choice of why this is at the top of the file-- this is the first piece of information youll need to know to even spawn the player and be able to use him.

Reply:
Player is different from other actors as you never manually spawn it so I thought it had a different system for setting it's params, now that I've checked, the params get set thru the same system as for the other actors that spawn with a scene so it makes sense. Should've checked before commenting.

Reply:
No problem! Yeah even though hes in a separate list from other actors, it's still the same entry.
The enum values will be used in scenes eventually, maybe when the new asset system exists and we can easily change things like that.

Maybe someday it will make sense to split player's header into different groups, but for now its fine here

Reply:
There's also a few use cases in code where this enums could be applied, such as part of the argument to `Play_SetupRespawnPoint`. 

In MM, we have `#define PLAYER_PARAMS(startBgCamIndex, initMode) ((startBgCamIndex & 0xFF) | ((initMode & 0xF) << 8))` (just rename initMode -> startMode) which you could introduce here, but it's also fine if you wanted to do it later

Reply:
ah I was having trouble even finding where that other param was read from. See it in z_play.c now.
Yeah, will merge this and then do a params followup